syntax = "proto3";
package control_relay_service.v1;

import "google/protobuf/empty.proto";

service ControlRelayHelloService {
  rpc Hello (HelloRequest) returns (HelloResponse) {}
}

service ControlRelayPacketService {
  rpc PacketTx (ControlRelayPacket) returns (google.protobuf.Empty) {}
  rpc ListenForPacketRx (google.protobuf.Empty) returns (stream ControlRelayPacket) {}  
}

message ControllerHello {
}

message DeviceHello {
  string device_name = 1;
}

message HelloRequest {
  oneof local_endpoint_hello {
    ControllerHello   controller = 1;
    DeviceHello       device = 2;
  }
}

message HelloResponse {
  oneof remote_endpoint_hello {
    ControllerHello   controller = 1;
    DeviceHello       device = 2;
  }
}

message ControlRelayPacket {
  string   device_name = 1;
  string   device_interface = 2;
  string   originating_rule = 3;
  bytes    packet = 10;
}
